ui/src/sources/components/SourceStep.tsx

```diff
@@ -207,22 +209,31 @@ class SourceStep extends PureComponent<Props, State> {
     setError(false)
   }
 
-  private syncHostnames = (sourceURLstring: string) => {
-    if (this.isEnterprise) {
-      if (sourceURLstring) {
-        const sourceURL = new URL(sourceURLstring)
-        const {source} = this.state
-        const metaserviceURL = new URL(source.metaUrl || DEFAULT_SOURCE.metaUrl)
-        if (sourceURL.hostname) {
-          metaserviceURL.hostname = sourceURL.hostname
-          this.onChangeInput('metaUrl')(metaserviceURL.href)
-        }
-      }
-    }
-  }
+  private handleSubmitUrl = async (sourceURLstring: string) => {
+    const {source} = this.state
+    const metaserviceURL = new URL(source.metaUrl || DEFAULT_SOURCE.metaUrl)
+    const sourceURL = new URL(sourceURLstring || DEFAULT_SOURCE.url)
+    metaserviceURL.hostname = sourceURL.hostname
 
-  private get isNewSource(): boolean {
-    return _.isNull(this.props.source)
+    if (isNewSource(source)) {
+      try {
+        let sourceFromServer = await createSource(source)
+        sourceFromServer = {...sourceFromServer, metaUrl: metaserviceURL.href}
+        this.props.addSource(sourceFromServer)
+        this.setState({
+          source: sourceFromServer,
+        })
+      } catch (err) {}
```

Comment:
does nothing need to happen if there is an err?

Reply:
There's no reason to do anything. they will get an actual error with consequences if/when they click next